packages/mattermost-redux/src/actions/search.ts

```diff
@@ -246,8 +246,10 @@ export function getPinnedPosts(channelId: string): ActionFunc {
 
             const profilesAndStatuses = getProfilesAndStatusesForPosts(result.posts, dispatch, getState);
             const missingChannels = dispatch(getMissingChannelsFromPosts(result.posts));
-            const arr: [Promise<any>, Promise<any>] = [profilesAndStatuses, missingChannels];
+            const updateLastViewedPinnedPostAt = Client4.updateLastViewedPinnedPostAt(channelId, getCurrentUserId(getState()));
+            const arr: [Promise<any>, Promise<any>, Promise<any>] = [profilesAndStatuses, missingChannels, updateLastViewedPinnedPostAt];
             await Promise.all(arr);
+            dispatch(getChannelStats(channelId));
```

Comment:
Do we actually need this call here? What if we were to have the reducer for the channel stats watch for the existing `RECEIVED_SEARCH_PINNED_POSTS` action and then recognize that it should set `has_new_pinned_posts` to false?

I know there's the above thread on needing to use the safer version of that API call here, but we might be able to remove this followup API call entirely

Reply:
@hmhealey
I think your idea can also achieve the goal.
I will reimplement this logic as you mentioned.

Reply:
@hmhealey 
I changed this to using RECEIVED_SEARCH_PINNED_POSTS to hide the badge.